recipes/ms-gsl/all/conanfile.py

```diff
@@ -50,8 +57,14 @@ def package_id(self):
         self.info.header_only()
 
     def package_info(self):
+        if int(self.version.split(".")[0]) > 2:
```

Comment:
It should be in `configure()` method instead. Also, do not parse self.version, use this helper:

    if tools.Version(self.version) >= "3.0.0":